pkg/gatherers/clusterconfig/operators_pods_and_events.go

```diff
@@ -282,28 +282,33 @@ func getContainerLogs(ctx context.Context,
 
 		logName := logFilename(c.Name, isPrevious)
 
-		// Fetch container logs and continue on error
-		logString, err := getContainerLogString(ctx, client, pod, c.Name, isPrevious, buf, &logTailLines)
+		// Fetch full container logs and continue on error
+		logString, err := getContainerLogString(ctx, client, pod, c.Name, isPrevious, buf, nil)
 		if err != nil {
 			klog.V(2).Infof("Error: %q", err)
 			continue
 		}
 
-		// check for stack tracer
+		// We need an array to slice the log
+		logArray := strings.Split(logString, "\n")
+
 		if found := stackTraceRegex.MatchString(logString); found {
 			klog.V(2).Infof(
 				"Stack trace found in log for %s container %s pod in namespace %s (previous: %v).",
 				c.Name,
 				pod.Name,
 				pod.Namespace,
 				isPrevious)
-			logString, err = getContainerLogString(ctx, client, pod, c.Name, isPrevious, buf, &logTailLinesLong)
-			if err != nil {
-				klog.V(2).Infof("Error: %q", err)
-				continue
-			}
+			logArray = getLogWithStacktracing(logArray)
 		}
 
+		// Set the log limits by number of lines
+		maxLines := int(logMaxLines)
+		if len(logArray) < maxLines {
+			maxLines = len(logArray)
+		}
+		logString = strings.Join(logArray[:maxLines], "\n")
```

Comment:
This seems wrong. We don't want beginning of the log.

Reply:
Oh, wait. I inverted the logic.